testdroid_cmdline.sh

```diff
@@ -298,6 +298,27 @@ function start_test_run {
 }
 
 ########################################
+# Get human readable name for device
+# Arguments:
+#   test_run_id
+#   device_run_id
+# Returns:
+#   String (human readable device id (or device_run_id if failure))
+#########################################
+function get_device_human_name {
+  test_run_id="$1"
+  device_run_id="$2"
+  test_run_item_url=$(url_from_template "${TD_TEST_RUN_ITEM_URL_TEMPLATE}" "${test_run_id}")
+  device_info_url="$test_run_item_url/device-runs/$device_run_id"
+  device_info_json=$(auth_curl "$device_info_url" --fail)
+  human_name=$(echo "$device_info_json" |jq -r '.deviceName + "-API\(.softwareVersion.apiLevel)"' |sed -e s/[^a-zA-Z0-9_-]/_/g)
```

Comment:
based on my tests, jq should return blank if field is missing. But you are right in that the filename will be a bit messy if there is no API-level for iOS. I will test to see what filename will look like.